firefox_ui_harness/mixins/base.py

```diff
@@ -39,9 +49,54 @@
     'focusmanager.testmode': True,
     'geo.provider.testing': True,
     'javascript.options.showInConsole': True,
+    'marionette.logging': False,
     'security.notification_enable_delay': 0,
     'signon.rememberSignons': False,
     'startup.homepage_welcome_url': 'about:blank',
     'toolkit.startup.max_resumed_crashes': -1,
     'toolkit.telemetry.enabled': False,
 }
+
+
+class FirefoxUIOptions(BaseMarionetteOptions):
+    def __init__(self, **kwargs):
+        BaseMarionetteOptions.__init__(self, **kwargs)
+
+        # Inheriting object must call this __init__ to set up option handling
+        self.add_option('--installer',
+                        dest='installer',
+                        help='installer of a Gecko application to use for running the tests')
+
+    def parse_args(self, *args, **kwargs):
+        options, tests = BaseMarionetteOptions.parse_args(self, *args, **kwargs)
+
+        # Bug 1142064 - We cannot easily extent options because registered handlers
+        # are called at the end in MarionetteBaseOptions.verify_usage(). As result it
+        # will abort due to no binary specified. Once the bug is fixed we can move
+        # the whole block to self.base_verify_usage().
+        if options.installer:
+            if options.binary:
+                self.error('Options --binary and --installer are mutually exclusive.')
+
+            options.binary = 'FAKED_VALUE'
+
+        if not tests:
+            tests = [firefox_puppeteer.manifest, firefox_ui_tests.manifest_all]
+
+        return (options, tests)
+
+
+class FirefoxUITestRunner(BaseMarionetteTestRunner):
+    def __init__(self, **kwargs):
+        BaseMarionetteTestRunner.__init__(self, **kwargs)
+
+        if not self.server_root:
+            self.server_root = firefox_ui_tests.resources
+
+        # Set default preferences for Firefox UI tests
+        # Bug 1141129 - Workaround necessary given that self.prefs defaults to None
+        prefs = self.prefs or {}
+        prefs.update(DEFAULT_PREFS)
+        self.prefs = prefs
```

Comment:
Our intention is to force preference values for our tests. Whenever something in Marionette changes it shouldn't totally bust our tests. In this case I agree, but if we can turn on e10s soon, I can remove this specific preference from our default list.